test/pathTools.cpp

```diff
@@ -190,6 +190,26 @@ TEST(pathTools, appendToDirectory)
             P5("a","b","c",".","foo.xml"));
 }
 
+TEST(pathTools, fileExists)
+{
+  ASSERT_TRUE(kiwix::fileExists(P3(".","test","example.zim")));
+  ASSERT_FALSE(kiwix::fileExists(P3(".","test","noFile.zim")));
+#ifdef __linux__
+  system("chmod -r ./test/example.zim");
+  ASSERT_TRUE(kiwix::fileExists(P3(".","test","example.zim")));
+  system("chmod +r ./test/example.zim");
+#endif
+}
```

Comment:
* If the test crashs before setting permission back, next run will have an error
* Actually there is an elegant way to change the permission, see https://stackoverflow.com/questions/592448/how-to-set-file-permissions-cross-platform-in-c (not a must... but would be nice, I let you decide what you want to do).

Reply:
Do you mean the `std::filesystem` answer of that link, I'll check that?  (the best chosen answer wouldn't work here I think, it cannot revoke permissions - there's also a similar `chmod` function in sys/stat.h but that also suffers from the same problem)